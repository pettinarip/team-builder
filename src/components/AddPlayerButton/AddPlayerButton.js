import React from 'react'
import PropTypes from 'prop-types'

const makeFakePlayer = () => {
  const number = Math.ceil(Math.random() * 100) + 1
  return {
    name: `Player ${number}`,
    number,
    icon: ''
  }
}

const AddPlayerButton = ({ onAddPlayer }) => (
  <div
    className='player add-player'
    title='Add new player'
    onClick={() => onAddPlayer(makeFakePlayer())}
  >
    <div className='player-number'>+</div>
  </div>
)

AddPlayerButton.propTypes = {
  onAddPlayer: PropTypes.func.isRequired
}

export default AddPlayerButton
