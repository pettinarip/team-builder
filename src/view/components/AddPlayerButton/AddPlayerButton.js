import React from 'react'
import PropTypes from 'prop-types'

const AddPlayerButton = ({ onClick }) => (
  <div
    className='player add-player item'
    title='Add new player'
    onClick={onClick}
  >
    <div className='player-number'>+</div>
  </div>
)

AddPlayerButton.propTypes = {
  onClick: PropTypes.func.isRequired
}

export default AddPlayerButton
