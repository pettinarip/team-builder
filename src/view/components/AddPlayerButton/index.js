import AddPlayerButton from './AddPlayerButton'

export default AddPlayerButton
