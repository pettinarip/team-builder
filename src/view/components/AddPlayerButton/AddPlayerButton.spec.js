import React from 'react'
import { shallow } from 'enzyme'
import AddPlayerButton from './AddPlayerButton'

describe('AddPlayerButton component', () => {
  it('should render the component properly', () => {
    const component = shallow(
      <AddPlayerButton onClick={() => null} />
    )

    expect(component).toMatchSnapshot()
  })

  it('should call onClick function when clicked', () => {
    const addPlayer = jest.fn()
    const component = shallow(
      <AddPlayerButton onClick={addPlayer} />
    )
    component.simulate('click')
    expect(addPlayer).toHaveBeenCalled()
  })
})
