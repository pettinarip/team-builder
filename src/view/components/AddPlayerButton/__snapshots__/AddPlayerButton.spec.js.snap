// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlayerButton component should render the component properly 1`] = `
<div
  className="player add-player item"
  onClick={[Function]}
  title="Add new player"
>
  <div
    className="player-number"
  >
    +
  </div>
</div>
`;
